app/controllers/users.rb

```diff
@@ -0,0 +1,28 @@
+get '/users' do
+  @users = User.all
+  erb :'/users/index'
+end
+
+get '/users/new' do
+  erb :'/users/new'
+end
+
+post '/users' do
+  @user = User.new(params[:user])
+  if @user.save
+    session[:user_id] = @user.id
```

Comment:
Uh oh: the rest of your app is setting the unique user id in `session[:id]`, not in `session[:user_id]`. Be sure to be consistent with this, since this is _very, very_ difficult to debug.